core/src/main/java/org/elasticsearch/action/admin/cluster/storedscripts/TransportGetStoredScriptAction.java

```diff
@@ -0,0 +1,77 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.storedscripts;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeReadAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptMetaData;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+public class TransportGetStoredScriptAction extends TransportMasterNodeReadAction<GetStoredScriptRequest,
+        GetStoredScriptResponse> {
+
+    @Inject
+    public TransportGetStoredScriptAction(Settings settings, TransportService transportService, ClusterService clusterService,
+                                          ThreadPool threadPool, ActionFilters actionFilters,
+                                          IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(settings, GetStoredScriptAction.NAME, transportService, clusterService, threadPool, actionFilters,
+                indexNameExpressionResolver, GetStoredScriptRequest::new);
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected GetStoredScriptResponse newResponse() {
+        return new GetStoredScriptResponse();
+    }
+
+    @Override
+    protected void masterOperation(GetStoredScriptRequest request, ClusterState state,
+                                   ActionListener<GetStoredScriptResponse> listener) throws Exception {
+        ScriptMetaData scriptMetadata = state.metaData().custom(ScriptMetaData.TYPE);
+        String result;
+        if (scriptMetadata != null) {
+            result = scriptMetadata.getScript(request.lang, request.id);
+        } else {
+            result = null;
+        }
+        listener.onResponse(new GetStoredScriptResponse(result));
```

Comment:
The PUT and DELETE actions both call into the ScriptService, should this not also call the ScriptService (`ScriptService.getStoredScript()`)?